otter/controller.py

```diff
@@ -303,6 +302,31 @@ def modify_state(group_info):
     return d
 
 
+@defer.inlineCallbacks
+def modify_and_trigger(dispatcher, group, logargs, modifier, *args, **kwargs):
+    """
+    Modify group state and trigger convergence after that
+
+    :param IScalingGroup group: Scaling group whose state is getting modified
+    :param log: Bound logger
+    :param modifier: Callable as described in IScalingGroup.modify_state
+
+    :return: Deferred with None
+    """
+    cannot_exec_pol_err = None
+    try:
+        yield group.modify_state(modifier, *args, **kwargs)
+    except CannotExecutePolicyError as ce:
+        cannot_exec_pol_err = ce
+    if tenant_is_enabled(group.tenant_id, config_value):
+        eff = Effect(
```

Comment:
`with_log` seems to be useful when specific keyword arguments are known. Here, since I already have all the args I thought of using `BoundFields` directly. Sorry I assumed it would be _cheaper_ to do this. Premature optimization :)